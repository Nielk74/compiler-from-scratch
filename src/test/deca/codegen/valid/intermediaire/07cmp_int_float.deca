{
    int x = 2;
    float y = 3.5;
    if (x < 3.5) {
        println(x, " < 3.5");
    }
    
    if (x < y) {
        println(x, " < ", y);
    }

    if (y != x) {
        println(y, " != ", x);
    }

    if (2.0 >= x) {
        println("2.0 >= ", x);
    }

    if (x <= 2.0) {
        println(x, " <= 2.0");
    }

    if (!(x > y || x > 2.1)){
        println(x, " < ", y, " && ", x, " < ", 2.1);
    }

    if (!(x > y && x < 2.1)){
        println(x, " > ", y, " || ", x, " < ", 2.1);
    }

    if (!(x < y && x > 2.1)){
        println(x, " < ", y, " || ", x, " > ", 2.1);
    }
}