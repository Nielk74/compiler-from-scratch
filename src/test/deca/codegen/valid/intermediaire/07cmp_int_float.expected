2 < 3.5
2 < 3.50000e+00
3.50000e+00 != 2
2.0 >= 2
2 <= 2.0
